packages/dds/merge-tree/src/mergeTree.ts

```diff
@@ -935,7 +933,12 @@ export class MergeTree {
                     refsToSlide.push(lref);
                 }
             } else {
-                segment.localRefs.removeLocalRef(lref);
+                // since we are pending, don't actually remove it,
+                // return the set of refs that will be removed on ack
+                //
+                if (!pending) {
```

Comment:
@Abe27342 I had to change this for rollback. it think it is safe, and will get cleaned up on acknowledgment. There is still some confusion in the code about when, why, and how we clear the segment of a local ref. do you see any potential issue with other local ref usages, like shared interval?

Reply:
AB#1845

Reply:
if ci passes i'll probably just commit this to unblock some other changes i have coming, but am open to making more changes in this area if we need to

Reply:
it seems reasonable. I don't think this change could cause issues to shared interval scheme, since that should always use one of `StayOnRemove` or `SlideOnRemove` and those cases are unaffected.

Reply:
the observation in that work item looks good, that model seems much nicer.